src/CompositeShape.cpp

```diff
@@ -43,6 +43,37 @@ bool CompositeShape::intersect(const Ray& r, float t0, float t1, double& distanc
     return false;
 }
 
+bool CompositeShape::intersect(const Vector3& p, const double radius) const {
+    if (!bb_.intersect(p, radius)) {
+        return false;
+    }
+    for(std::vector<std::pair<ShapePtr, Transform> >::const_iterator it = shapes_.begin(); it != shapes_.end(); ++it) {
```

Comment:
Can you use the 'auto' statement? According to your CMakeLists you should be able to use it

Reply:
I'm not familiar with auto. Did a quick Google.
You mean something like for(auto it =shapes.begin()...?
I might prefer it like this. It makes it clearer what data type you are using.

Reply:
Yup that's what I meant. Not using it seems a bit outdated, read, e.g., https://subscription.packtpub.com/book/application_development/9781786465184/1/ch01lvl1sec5/using-auto-whenever-possible . Personally, I find it even harder to read if it's a const iterator kind of type